src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamConformanceTests.cs

```diff
@@ -30,6 +33,11 @@ protected override async Task<StreamPair> CreateWrappedConnectedStreamsAsync(Str
                 ssl2.AuthenticateAsServerAsync(cert, false, GetSslProtocols(), false)
             }.WhenAllOrAnyFailed().ConfigureAwait(false);
 
+            if (wrapped.Stream2 is NetworkStream)
+            {
+                networkStream = wrapped.Stream2 as NetworkStream;
+            }
+
```

Comment:
Sorry I wasn't clear.  I actually meant doing the read/writes here.  No other test code should need to change.  Right?

Reply:
That may be tricky as I need to shutdown one side and this method is shared for multiple tests. Without the shutdown I don't know if the message was processed since there is not visible output. e.g. I read till EOF now.

Reply:
I'm not following; can you elaborate?  I thought the problem was that authentication may have resulted in an extra message being sent that wasn't necessarily read.  If so, why can't this method be:
```C#
        protected override async Task<StreamPair> CreateWrappedConnectedStreamsAsync(StreamPair wrapped, bool leaveOpen = false)
        {
            X509Certificate2? cert = Test.Common.Configuration.Certificates.GetServerCertificate();
            var ssl1 = new SslStream(wrapped.Stream1, leaveOpen, delegate { return true; });
            var ssl2 = new SslStream(wrapped.Stream2, leaveOpen, delegate { return true; });

            await new[]
            {
                ssl1.AuthenticateAsClientAsync(cert.GetNameInfo(X509NameType.SimpleName, false), null, GetSslProtocols(), false),
                ssl2.AuthenticateAsServerAsync(cert, false, GetSslProtocols(), false)
            }.WhenAllOrAnyFailed().ConfigureAwait(false);

            ssl1.Write(new byte[1]);
            ssl2.Write(new byte[1]);
            Assert.Equal(1, ssl2.Read(new byte[1]));
            Assert.Equal(1, ssl1.Read(new byte[1]));

            return new StreamPair(ssl1, ssl2);
        }
```
?

Reply:
yes, adding read/write from both sides should work IMHO. Do you think I should do it only for TLS 1.3 or for protocol versions? (For others it should not be needed but it does not hurt either)

Reply:
If it's not needed for earlier TLS's, let's just do it for 1.3.  Thanks.

Reply:
updated.